src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -333,6 +331,101 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 }
 
 
+/*
+ * CreateCitusLocalTablePartitionOf generates and executes the necessary commands
+ * to create a table as partition of a partitioned Citus Local Table.
+ * The conversion is done by CreateCitusLocalTable.
+ */
+void
+CreateCitusLocalTablePartitionOf(CreateStmt *createStatement, Oid relationId,
+								 Oid parentRelationId)
+{
+	if (createStatement->partspec)
```

Comment:
I think it might worth to comment why checking `partspec` is useful, so that other developers don't spend time looking to see what it means in postgres code:

```c
if (createStatement->partspec)
{
	/*
	 * Since partspec represents "PARTITION BY" clause, being different than
	 * NULL means that given CreateStmt attempts to create a parent table
	 * at the same time. That means multi-level partitioning within this
	 * function's context.
	 */
```